google/oauth2/service_account.py

```diff
@@ -639,6 +640,31 @@ def with_target_audience(self, target_audience):
             quota_project_id=self.quota_project_id,
         )
 
+    def _with_use_iam_endpoint(self, use_iam_endpoint):
```

Comment:
Is there a scenario where the IAM endpoint needs to be overridden? E.g. mtls, TPC?